source/PlanetLabel.cpp

```diff
@@ -91,76 +82,135 @@ PlanetLabel::PlanetLabel(const Point &position, const StellarObject &object, con
 	else if(planet.GetGovernment())
 	{
 		government = "(" + planet.GetGovernment()->GetName() + ")";
-		color = planet.GetGovernment()->GetColor();
-		color = Color(color.Get()[0] * .5f + .3f, color.Get()[1] * .5f + .3f, color.Get()[2] * .5f + .3f);
+		color = Color::Combine(.5f, planet.GetGovernment()->GetColor(), 1.f, Color(.3f));
 		if(!planet.CanLand())
 			hostility = 3 + 2 * planet.GetGovernment()->IsEnemy();
 	}
 	else
 	{
-		color = Color(.3f, .3f, .3f, 1.f);
 		government = "(No government)";
+		color = Color(.3f);
 	}
-	float alpha = static_cast<float>(min(.5, max(0., .6 - (position.Length() - object.Radius()) * .001 * zoom)));
-	color = Color(color.Get()[0] * alpha, color.Get()[1] * alpha, color.Get()[2] * alpha, 0.);
-
-	if(!system)
-		return;
-
-	// Figure out how big the label has to be.
-	double width = max(FontSet::Get(18).Width(name), FontSet::Get(14).Width(government)) + 8.;
 
-	// Try to find a label direction that not overlapping under any zoom.
-	for(int d = 0; d < 4; ++d)
-		if(!Overlaps(*system, object, Preferences::MinViewZoom(), width, d)
-				&& !Overlaps(*system, object, Preferences::MaxViewZoom(), width, d))
+	// Figure out how big the label is.
+	const Font &font = FontSet::Get(14);
+	const Font &bigFont = FontSet::Get(18);
+	const double labelWidth = max(bigFont.Width(name), font.Width(government));
+	const double nameHeight = bigFont.Height();
+	const double labelHeight = nameHeight + (government.empty() ? 0. : 1. + font.Height());
+	const Point labelDimensions = {labelWidth + BORDER * 2., labelHeight + BORDER * 2.};
+
+	// Try to find a label direction that is not overlapping under any zoom.
+	const double minZoom = Preferences::MinViewZoom();
+	const double maxZoom = Preferences::MaxViewZoom();
+	for(const double angle : LINE_ANGLES)
+	{
+		SetBoundingBox(labelDimensions, angle);
+		if(!HasOverlaps(labels, system, object, minZoom) && !HasOverlaps(labels, system, object, maxZoom))
```

Comment:
It probably makes sense to combine these into one function, to avoid iterating twice through the lists.

Reply:
I kinda want to leave room for checking more zoom values though, rather than hardcoding them into HasOverlaps.

Reply:
In fact, for exactly the kind of case you bring up above!  That overlap doesn't occur at min or max zoom, but at 3 of the mid-range zooms.